mods/default/tools.lua

```diff
@@ -24,383 +24,242 @@ minetest.register_item(":", {
-	groups = {sword = 1, flammable = 2}
-})
-
-minetest.register_tool("default:sword_stone", {
-	description = S("Stone Sword"),
-	inventory_image = "default_tool_stonesword.png",
-	tool_capabilities = {
-		full_punch_interval = 1.2,
-		max_drop_level=0,
-		groupcaps={
-			snappy={times={[2]=1.4, [3]=0.40}, uses=20, maxlevel=1},
-		},
-		damage_groups = {fleshy=4},
+	stone = {
+		desc = S("Stone Sword"),
+		punch = 1.2, drop = 0, damage = 4,
+		cap = {times={[2]=1.4, [3]=0.40}, uses=20, maxlevel=1},
+		groups = {sword = 1},
 	},
-	sound = {breaks = "default_tool_breaks"},
-	groups = {sword = 1}
-})
-
-minetest.register_tool("default:sword_bronze", {
-	description = S("Bronze Sword"),
-	inventory_image = "default_tool_bronzesword.png",
-	tool_capabilities = {
-		full_punch_interval = 0.8,
-		max_drop_level=1,
-		groupcaps={
-			snappy={times={[1]=2.75, [2]=1.30, [3]=0.375}, uses=25, maxlevel=2},
-		},
-		damage_groups = {fleshy=6},
+	bronze = {
+		desc = S("Bronze Sword"),
+		punch = 0.8, drop = 1, damage = 6,
+		cap = {times={[1]=2.75, [2]=1.30, [3]=0.375}, uses=25, maxlevel=2},
+		groups = {sword = 1},
 	},
-	sound = {breaks = "default_tool_breaks"},
-	groups = {sword = 1}
-})
-
-minetest.register_tool("default:sword_steel", {
-	description = S("Steel Sword"),
-	inventory_image = "default_tool_steelsword.png",
-	tool_capabilities = {
-		full_punch_interval = 0.8,
-		max_drop_level=1,
-		groupcaps={
-			snappy={times={[1]=2.5, [2]=1.20, [3]=0.35}, uses=30, maxlevel=2},
-		},
-		damage_groups = {fleshy=6},
+	steel = {
+		desc = S("Steel Sword"),
+		punch = 0.8, drop = 1, damage = 6,
+		cap = {times={[1]=2.5, [2]=1.20, [3]=0.35}, uses=30, maxlevel=2},
+		groups = {sword = 1},
 	},
-	sound = {breaks = "default_tool_breaks"},
-	groups = {sword = 1}
-})
-
-minetest.register_tool("default:sword_mese", {
-	description = S("Mese Sword"),
-	inventory_image = "default_tool_mesesword.png",
-	tool_capabilities = {
-		full_punch_interval = 0.7,
-		max_drop_level=1,
-		groupcaps={
-			snappy={times={[1]=2.0, [2]=1.00, [3]=0.35}, uses=30, maxlevel=3},
-		},
-		damage_groups = {fleshy=7},
+	mese = {
+		desc = S("Mese Sword"),
+		punch = 0.7, drop = 1, damage = 7,
+		cap = {times={[1]=2.0, [2]=1.00, [3]=0.35}, uses=30, maxlevel=3},
+		groups = {sword = 1},
 	},
-	sound = {breaks = "default_tool_breaks"},
-	groups = {sword = 1}
-})
+	diamond = {
+		desc = S("Diamond Sword"),
+		punch = 0.7, drop = 1, damage = 8,
+		cap = {times={[1]=1.90, [2]=0.90, [3]=0.30}, uses=40, maxlevel=3},
+		groups = {sword = 1},
+	}
+}
 
-minetest.register_tool("default:sword_diamond", {
-	description = S("Diamond Sword"),
-	inventory_image = "default_tool_diamondsword.png",
-	tool_capabilities = {
-		full_punch_interval = 0.7,
-		max_drop_level=1,
-		groupcaps={
-			snappy={times={[1]=1.90, [2]=0.90, [3]=0.30}, uses=40, maxlevel=3},
+for mat, tool in pairs(tool_sword) do
+	minetest.register_tool("default:sword_".. mat, {
+		description = tool.desc,
```

Comment:
**description = S(tool.desc),**

Reply:
I had those in the tables, but you are right, its less redundant in the loops, thanks!